src/Fl_Group.cxx

```diff
@@ -456,12 +511,25 @@ void Fl_Group::insert(Fl_Widget &o, int index) {
   if (o.parent()) {
     Fl_Group* g = o.parent();
     int n = g->find(o);
-    if (g == this) {
+    if (g == this) { // avoid expensive remove() and add() if we just move a widget within the group
       if (index > n) index--;
-      if (index == n) return;
+      index = on_move(n, index);
+      if (index == n) return; // this includes (children_ == 1)
```

Comment:
The order of the above two statements implies that we would call `on_move(n, n)` if `index == n`, i.e. we would call the subclass for a no-op "move". Should we really do this?

Reordering lines 516/517 would avoid this issue.

I'm wondering when calling `on_move(n, n)` would be useful but maybe there is a reason?

Reply:
It's on purpose. Not sure what it may be good for, but I figured I want to capture all attempts to move.